### DISCUSSION TITLE
Error installing platformio-ide-terminal

### DISCUSSION THREAD

**Discussion by chrisungarro** [community/community#22891] [New to GitHub]
<pre><code>Please i have been getting the error below when installing platformio-ide-terminal packages..

it just stops and displays error below i have tried a number of times and same error always.

i need help to solve this problem.

thanks in anticipation.

gyp info it worked if it ends with ok
gyp info using node-gyp@3.4.0
gyp info using node@6.9.5 | win32 | ia32
gyp http GET https://atom.io/download/electron/v2.0.5/iojs-v2.0.5.tar.gz
gyp http 200 https://atom.io/download/electron/v2.0.5/iojs-v2.0.5.tar.gz
gyp http GET https://atom.io/download/electron/v2.0.5/SHASUMS256.txt
gyp http GET https://atom.io/download/electron/v2.0.5/win-x86/iojs.lib
gyp http GET https://atom.io/download/electron/v2.0.5/win-x64/iojs.lib
gyp http 200 https://atom.io/download/electron/v2.0.5/SHASUMS256.txt
gyp http 200 https://atom.io/download/electron/v2.0.5/win-x64/iojs.lib
gyp WARN install got an error, rolling back install
gyp http 200 https://atom.io/download/electron/v2.0.5/win-x86/iojs.lib
gyp ERR! install error
gyp ERR! stack Error: EPERM: operation not permitted, open 'C:\Users\learn\.atom\.node-gyp\.node-gyp\iojs-2.0.5\x64\iojs.lib'
gyp ERR! stack at Error (native)
gyp ERR! System Windows_NT 6.1.7601
gyp ERR! command "C:\\Users\\learn\\AppData\\Local\\atom\\app-1.28.2\\resources\\app\\apm\\bin\\node.exe" "C:\\Users\\learn\\AppData\\Local\\atom\\app-1.28.2\\resources\\app\\apm\\node_modules\\node-gyp\\bin\\node-gyp.js" "install" "--runtime=electron" "--target=2.0.5" "--dist-url=https://atom.io/download/electron" "--arch=ia32" "--ensure"
gyp ERR! cwd C:\Users\learn\.atom
</code></pre>

**Comment by lee-dohm** [comment, marked as answer]
<p>Some packages require build tools to install them because they require native components to be built. On Windows, this requires installing tools that are described in <a href="https://flight-manual.atom.io/hacking-atom/sections/hacking-on-atom-core/#building" rel="nofollow noopener">the Atom Flight Manual section on Building Atom.</a> Once you install those tools, then try reinstalling the package. If that doesn’t work, you may want to <a href="https://github.com/platformio/platformio-atom-ide-terminal/issues" rel="nofollow noopener">open an issue on the package itself</a>.</p>
<p>I hope that helps!</p>
